src/main/java/seedu/sherpass/util/Reminder.java

```
@@ -41,44 +41,52 @@ public void showReminders() {
+                for (Task task : filteredDailyTasks) {
+                    System.out.println(task.toString());
+                }
             }
+            ui.showLine();
+        } catch (Exception e) {
+            System.out.println("Wrong date format");
         }
-        ui.showLine();
     }
 
     /**
      * Prints tasks that is due in the current week
      * with reference to user local machine date.
      */
     public void showWeeklyTask() {
-        LocalDate nextWeekDate = currentDate.plusDays(7);
+        try {
+            LocalDate nextWeekDate = currentDate.plusDays(7);
 
-        ArrayList<Task> filteredThisWeekTasks = (ArrayList<Task>) tasks.stream()
-                .filter((t) -> isBeforeDate(t.getByDate(), nextWeekDate))
-                .filter((t) -> !t.isDone())
-                .sorted(Comparator.comparing(Task::getByDate))
-                .collect(toList());
+            ArrayList<Task> filteredThisWeekTasks = (ArrayList<Task>) tasks.stream()
+                    .filter((t) -> isBeforeDate(t.getByDate(), nextWeekDate))
+                    .filter((t) -> !t.isDone())
+                    .sorted(Comparator.comparing(Task::getByDate))
+                    .collect(toList());
 
-        if (filteredThisWeekTasks.isEmpty()) {
-            ui.showToUser("You do not have any pending task for the week.");
-        } else {
-            System.out.println("Tasks to be done within the week:");
-            for (Task task : filteredThisWeekTasks) {
-                System.out.println(task.toString());
+            if (filteredThisWeekTasks.isEmpty()) {
+                ui.showToUser("You do not have any pending task for the week.");
+            } else {
+                System.out.println("Tasks to be done within the week:");
+                for (Task task : filteredThisWeekTasks) {
+                    System.out.println(task.toString());
+                }
             }
+            ui.showLine();
+        } catch (Exception e) {
```

Do put in the specific exception here too.

---

It is a temporary fix to catch all exceptions to prevent the application from crashing. I know it is a bad practice. I will have a proper fix later.